src/runtime/recipe/handle.ts

```diff
@@ -376,4 +384,9 @@ export class Handle implements Comparable<Handle> {
   findConnectionByDirection(dir: Direction): HandleConnection|undefined {
     return this._connections.find(conn => conn.direction === dir);
   }
+
+  associateHandle(handle: Handle) {
```

Comment:
Is there a way to assert ordering of associated handles?
My concern is that the tuple might accidentally get reordered.